cartographer_ros/cartographer_ros/map_builder_bridge.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2016 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <memory>
+#include <unordered_set>
+
+#include "cartographer/mapping/map_builder.h"
+#include "cartographer_ros/node_options.h"
+#include "cartographer_ros/sensor_bridge.h"
+#include "cartographer_ros/tf_bridge.h"
+#include "cartographer_ros_msgs/FinishTrajectory.h"
+#include "cartographer_ros_msgs/SubmapEntry.h"
+#include "cartographer_ros_msgs/SubmapList.h"
+#include "cartographer_ros_msgs/SubmapQuery.h"
+
+namespace cartographer_ros {
+
+class MapBuilderBridge {
+ public:
+  MapBuilderBridge(const NodeOptions& options,
+                   const std::unordered_set<string>& expected_sensor_ids,
+                   tf2_ros::Buffer* tf_buffer);
+  ~MapBuilderBridge();
+
+  MapBuilderBridge(const MapBuilderBridge&) = delete;
+  MapBuilderBridge& operator=(const MapBuilderBridge&) = delete;
+
+  void HandleOdometryMessage(const string& sensor_id,
```

Comment:
As the TODO states, I'd rather not have accessors for any of the bridges. This one was simple to avoid for now.